src/lwc/geGiftEntryFormApp/geGiftEntryFormApp.js

```diff
@@ -1,37 +1,396 @@
+    * @description Get the value for DataImport__c.Payment_Declined_Reason__c from
+    * the purchase call response.
+    *
+    * @param {object} response: Http response object
+    *
+    * @return {string}: Reason the payment was declined
+    */
+    getPaymentDeclinedReason(response) {
+        const isSuccessfulPurchase = response.statusCode === 201;
+        return isSuccessfulPurchase ? null : JSON.stringify(response.body);
+    }
+
+    /*******************************************************************************
+    * @description Get the message or errors from a failed purchase call.
+    *
+    * @param {object} response: Http response object
+    *
+    * @return {string}: Message from a failed purchase call response
+    */
+    getFailedPurchaseMessage(response) {
+        // For some reason the key in the body object for 'Message'
+        // in the response we receive from Elevate is capitalized.
+        // Also checking for lowercase M in message in case they fix it.
+        return response.body.Message ||
+            response.body.message ||
+            response.body.errors.map(error => error.message).join(', ') ||
+            this.CUSTOM_LABELS.commonUnknownError;
+    }
+
+    /*******************************************************************************
+    * @description Sends the Data Import into BDI for processing and navigates to
+    * the opportunity record detail page on success.
+    *
+    * @param {object} dataImportRecord: A DataImport__c record
+    * @param {boolean} hasUserSelectedDonation: True if a selection had been made in
+    * the 'Review Donations' modal.
+    * Determines BDI matching criteria.
+    *   true = "single match or create" and means we are updating
+    *   false = "do not match"
+    */
+    processDataImport = async () => {
+        this.loadingText = this.CUSTOM_LABELS.geTextProcessing;
+
+        submitDataImportToBDI({ diRecord: this.dataImportRecord, updateGift: this.hasUserSelectedDonation })
+            .then(opportunityId => {
+                this.loadingText = this.CUSTOM_LABELS.geTextNavigateToOpportunity;
+                this.navigateToRecordPage(opportunityId);
+            })
             .catch(error => {
-                event.detail.error(error);
+                // TODO: Placeholder method for handling the scenario in which
+                // we've charged a card successfully, but BDI processing failed.
+                this.doSomethingLoud(error);
             });
     }
 
+    // TODO: Placeholder method, remove/replace in work item to better handle purchase call errors
+    doSomethingLoud(error) {
+        this.errorCallback(error);
+        // TODO: Potentially have to check for other types of status that MAY
+        // indicate a charge could still occur (pending, authorized, etc)
+        if (this.dataImportRecord[PAYMENT_STATUS__C] === PAYMENT_TRANSACTION_STATUS_ENUM.CAPTURED) {
+            showToast('HEY!',
```

Comment:
@melissabarber - this exact behavior is being defined still, but FYI we will need to figure out the exact messaging for this scenario at some point. I'd love to start the conversation but I think we'll have to resolve it in a future sprint but just wanted to keep you in the loop!